staging/src/k8s.io/api/networking/v1/types.go

```diff
@@ -201,3 +201,300 @@ type NetworkPolicyList struct {
 	// Items is a list of schema objects.
 	Items []NetworkPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Ingress is a collection of rules that allow inbound connections to reach the
+// endpoints defined by a backend. An Ingress can be configured to give services
+// externally-reachable urls, load balance traffic, terminate SSL, offer name
+// based virtual hosting etc.
+type Ingress struct {
```

Comment:
I expected v1 Ingress and IngressClass types to be added to:
* `staging/src/k8s.io/api/networking/v1/register.go#addKnownTypes`
* `pkg/registry/networking/rest.RESTStorageProvider#v1Storage`
* `test/integration/etcd/data.go`